Adding HeroCarousel component

gvn opened the pull request:
Closes #22

Pomax commented:
technical nit: is this a true "hero" unit in the sense of being a massive banner across the top of a page, or is this a standard carousel?

Pomax commented on src/components/hero-carousel/hero-carousel.jsx:
```diff
@@ -0,0 +1,52 @@
+      let classNames = `hero${this.state.activeTrigger === index ? ` active` : ``}`;
+
+      return (
+        <button key={index} className={classNames} onClick={this.onTriggerClick.bind(null, index)}>
```
`onClick={index => this.onTriggerClick(index)}`?

gvn replied:
What's the advantage of doing it that way?

Pomax replied:
relying on syntax rather than low level JS functions, mainly. The arrow notation does "out of the box" what this low level spec function call with null-context argument is used for, so it's going to be easier to understand (once you're familiar with it, it's easy to forget that it's a pretty hardcore "you should never need this in normal code" function)

gvn commented:
I was thinking "heroes" as in the people, but I'm open to calling it something else if you have suggestions. It's just for our internal reference, so not super important...

Pomax commented:
Given that "hero" has a meaning in design, probably best not to overload it (also, "heroes" is a pretty English term).  I'd say we stick with the "name it after what it does" philosophy that got us the mofo-ui name, so just `Carousel` should work I think?

gvn commented:
Ok, I renamed to `ThumbCarousel` since I feel like it's a variation on a typical carousel and I don't want to make `ActualCarousel` later.

I changed the handler to a single method reference to prevent repeatedly spinning up new function instances. Slight performance win...

Pomax commented:
:+1: on the name. Good thinking with the `this.onTriggerClick`, too. R+ from me!

gvn commented:
![hi5](https://cloud.githubusercontent.com/assets/206379/17264068/840b8906-5599-11e6-8f86-4efe24d68fda.gif)
